omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/DiagnosisResource2_2.java

```diff
@@ -0,0 +1,220 @@
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#getGETModel(Representation)
+	 */
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl model = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			model
+			        .property("uuid", new StringProperty())
+			        .property("diagnosis", new StringProperty())
+			        .property("condition", new StringProperty())
+			        .property("certainty", new EnumProperty(ConditionVerificationStatus.class))
+			        .property("rank", new IntegerProperty())
+			        .property("patient", new RefProperty("#/definitions/PatientGetRef"))
+			        .property("voided", new BooleanProperty());
+		}
+		return model;
+	}
+	
+	/**
+	 * @param diagnosis
+	 * @return Diagnosis's name
+	 */
+	@PropertyGetter("display")
+	public String getDisplayString(Diagnosis diagnosis) {
+		if (diagnosis.getDiagnosis() == null) {
+			return "";
+		} else {
+			if (diagnosis.getDiagnosis().getCoded() != null) {
+				return diagnosis.getDiagnosis().getCoded().getName().getName();
+			}
+			return diagnosis.getDiagnosis().getNonCoded();
+		}
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		
+		description.addRequiredProperty("diagnosis");
+		description.addRequiredProperty("encounter");
+		description.addRequiredProperty("condition");
+		description.addRequiredProperty("certainty");
+		description.addRequiredProperty("patient");
+		description.addRequiredProperty("rank");
+		
+		return description;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#getCREATEModel(Representation)
+	 */
+	@Override
+	public Model getCREATEModel(Representation rep) {
+		
+		return new ModelImpl()
+		        .property("diagnosis", new StringProperty())
+		        .property("encounter", new StringProperty())
+		        .property("condition", new StringProperty())
+		        .property("certainty", new StringProperty())
+		        .property("patient", new StringProperty().example("uuid"))
+		        .property("rank", new IntegerProperty());
+		
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getUpdatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("diagnosis");
+		description.addRequiredProperty("encounter");
```

Comment:
I do not think we should support updating the encounter

Reply:
Sure. On it